### PR TITLE
Fix the basic auth client authoriation header

### PR DESCRIPTION
**Pull request by er1c**
This matches the behavior of the digest auth `makeDigestAuthorization`  I tried to look for other references..specifically I'm not sure what: https://github.com/frugalmechanic/fm-http/blob/bd8ca492e883de4968d806ac4b6c7d926fa4b5fa/src/main/scala/fm/http/Headers.scala#L745-L748 does

### CODE REVIEW

**Comment by tpunder** [comment]
It doesn't look like it does anything.  Should either fix that or remove it

**Comment by er1c** [comment]
I'll attempt to add a test in too

**Comment by er1c** [comment]
PR updated

File: src/test/scala/fm/http/TestClientAndServer.scala [hunk 1]
```diff
@@ -70,7 +70,7 @@ object TestClientAndServer {
   private val UTF8Header: Headers   = Headers(("Content-Type", "text/html; charset=utf-8"))
   private val Latin1Header: Headers = Headers(("Content-Type", "text/html; charset=ISO-8859-1"))
 
-  protected val unwrappedHandler: PartialFunction[Request, Response] = {
+  protected def unwrappedHandler: PartialFunction[Request, Response] = (request: Request) => request match {
```
Comment (tpunder) [inline src/test/scala/fm/http/TestClientAndServer.scala]: Why was this changed from a `val` to a `def`?

File: src/test/scala/fm/http/TestClientAndServer.scala [hunk 2]
```diff
@@ -357,6 +364,11 @@ final class TestClientAndServer extends FunSuite with Matchers with BeforeAndAft
     getSync("/redirect6", 302, "/redirect5", clientNoFollowRedirects)
   }
 
+  test("Authentication - Basic Auth") {
+    getSync("/basic_auth", 401, "You need a valid user and password to access this content.")
+    getSync("/basic_auth", 200, "ok", client.withBasicAuth("foo", "bar"))
```
Comment (tpunder) [inline src/test/scala/fm/http/TestClientAndServer.scala]: What about testing for invalid credentials?

**Comment by er1c** [comment]
PR updated for typos & bad credentials case

File: src/test/scala/fm/http/TestClientAndServer.scala [hunk 3]
```diff
@@ -357,6 +364,12 @@ final class TestClientAndServer extends FunSuite with Matchers with BeforeAndAft
     getSync("/redirect6", 302, "/redirect5", clientNoFollowRedirects)
   }
 
+  test("Authentication - Basic Auth") {
+    getSync("/basic_auth", 401, "You need a valid user and password to access this content.")
+    getSync("/basic_auth", 401, "You need a valid user and password to access this content.", client.withBasicAuth("bar", "foo"))
+    getSync("/basic_auth", 200, "ok", client.withBasicAuth("foo", "bar"))
+  }
```
Comment (tpunder) [inline src/test/scala/fm/http/TestClientAndServer.scala:371]: This would be more useful if the server side piece of this made use of `fm.http.server.BasicAuth` to test the server side implementation instead of the manually coded `handleBasicAuth` method.

Comment (er1c) [reply src/test/scala/fm/http/TestClientAndServer.scala:371]: I looked into it, but it wasn't obvious to me howto do something quickly.  Feel free to fix it up.  The client header was the more important bug I was trying to verify/fix for `ta-webservice-api`
